plaso/storage/sqlite/sqlite_file.py

```diff
@@ -431,6 +431,78 @@ def _ReadAndCheckStorageMetadata(self, check_readable_only=False):
     self.serialization_format = metadata_values['serialization_format']
     self.storage_type = metadata_values['storage_type']
 
+  def _UpdateEventDataIdentifierAfterDeserialize(self, event):
+    """Updates the event data identifier of an event after deserialization.
+
+    Args:
+      event (EventObject):
+    """
+    row_identifier = getattr(event, 'event_data_row_identifier', None)
+    if row_identifier is not None:
+      event_data_identifier = identifiers.SQLTableIdentifier(
+          'event_data', row_identifier)
+      event.SetEventDataIdentifier(event_data_identifier)
+
+      delattr(event, 'event_data_row_identifier')
+
+  def _UpdateEventDataIdentifierBeforeSerialize(self, event):
+    """Updates the event data identifier of an event before serialization.
+
+    Args:
+      event (EventObject):
+
+    Raises:
+      IOError: if the event data identifier type is not supported.
+      OSError: if the event data identifier type is not supported.
+    """
+    event_data_identifier = event.GetEventDataIdentifier()
+
+    # TODO: change to no longer allow event_data_identifier is None
+    # after refactoring every parser to generate event data.
+    if event_data_identifier:
+      if not isinstance(event_data_identifier, identifiers.SQLTableIdentifier):
+        raise IOError('Unsupported event data identifier type: {0!s}'.format(
+            type(event_data_identifier)))
+
+      setattr(event, 'event_data_row_identifier',
+              event_data_identifier.row_identifier)
+
+  def _UpdateEventIdentifierAfterDeserialize(self, event_tag):
+    """Updates the event identifier of an event after deserialization.
+
+    Args:
+      event_tag (EventTag): event tag.
+
+    Raises:
+      ValueError: if the event row identifier attribute is missing.
+    """
+    row_identifier = getattr(event_tag, 'event_row_identifier', None)
+    if row_identifier is None:
+      raise ValueError('Missing event row identifier attribute')
+
+    event_identifier = identifiers.SQLTableIdentifier(
+        self._CONTAINER_TYPE_EVENT, row_identifier)
+    event_tag.SetEventIdentifier(event_identifier)
+
+    delattr(event_tag, 'event_row_identifier')
+
+  def _UpdateEventIdentifierBeforeSerialize(self, event_tag):
+    """Updates the event identifier of an event before serialization.
```

Comment:
This doesn't update update event_data_identifier, it sets the event_data_row_identifier on the event tag. Please update the docstring.

Reply:
done